scripts/skills/special/rf_poise_effect.nut

```diff
@@ -0,0 +1,142 @@
+/*
+Internally we calculate the current threshold inversely. That makes it much cleaner in code to handle increases and reduction of PoiseMax.
+Our intended behavior is that a change in PoiseMax also changes the Poise of that character by the same amount:
+
+The PoiseMax is purely handled in the Properties while the current Poise is handled in this skill.
+It is much cleaner when a change to PoiseMax can solely focus on changing variables in Properties.
+*/
+
+this.rf_poise_effect <- ::inherit("scripts/skills/skill", {
+	m = {
+		// Config
+		HeadshotMult = 1.5,	// multiplier for incoming poise damage on a hit to the head (unless steelbrow was learned)
+
+		// Private
+		InversePoise = 0,	// This is the inverse of Poise and counts upwards instead of downwards as Poise damaged is received
+	},
+
+	function create()
+	{
+		this.m.ID = "effects.poise";
+		this.m.Name = "Poise";
+		this.m.Description = ::Reforged.Mod.Tooltips.parseString("Certain skills, like [stunning attacks|Concept.StunningAttack], damage your [Poise|Concept.Poise] and apply an effect when they reduce it to 0 or below. If that happens or at the start of your turn your Poise is reset back to your maximum Poise.");
+		this.m.Icon = "ui/traits/trait_icon_15.png";	// Placeholder - TODO: create unique icon instead of usin
+		this.m.Type = ::Const.SkillType.StatusEffect;
+		this.m.Order = ::Const.SkillOrder.Any;
+		this.m.IsActive = false;
+		this.m.IsHidden = false;
+		this.m.IsSerialized = false;
+		this.m.IsRemovedAfterBattle = false;
+	}
+
+	function getTooltip()
+	{
+		local tooltip = this.skill.getTooltip();
+		return tooltip;
+	}
+
+	function getName()
+	{
+		if (::MSU.isNull(this.getContainer())) return this.m.Name;
+
+		local properties = this.getContainer().getActor().getCurrentProperties();
+		local suffix = properties.IsImmuneToStunFromPoise ? " (Immune)" : " (" + this.getPoise(properties) + " / " + properties.getPoiseMax() + ")";
+
+		return this.m.Name + suffix;
+	}
+
+	function onTurnStart()
+	{
+		this.resetPoise();
+	}
+
+	function onBeforeDamageReceived( _attacker, _skill, _hitInfo, _properties )
+	{
+		if (_skill == null || !_skill.m.IsDamagingPoise) return;
+
+		local attackPower = this.calculatePoiseDamage(_attacker, _skill, _hitInfo.BodyPart == ::Const.BodyPart.Head);
+		if (attackPower == 0) return;
+
+		this.onPoiseDamageReceived(_attacker, _skill, attackPower, true, _properties);
```

Comment:
If we refactor it to apply poise damage from the attacker's perspective then this logic needs to be rewritten and moved to `onTargetHit`.